### PR TITLE
Update gulp to version 4

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Update gulp to version 4

Update build steps to follow new patterns

Remove run-sequence

[stage-18]

## Motivation and Context
Update dependency version

## How Has This Been Tested?
e2e tests pass, tested locally and on staging. Tested various other steps like ch-build.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@ezequielc please review. Thanks!

**Review by ezequielc — APPROVED** [review]

File: gulpfile.js [hunk 1]
```diff
@@ -326,31 +323,29 @@ gulp.task("test:e2e:core", ["test:webdriver_update"],factory.testE2EAngular({
...
 });
 
-gulp.task('dev', ['lint', 'build-pieces', 'browser-sync', 'watch']);
+exports.dev = gulp.parallel('lint', 'build-pieces', 'browser-sync', 'watch');
 
 /**
  * Default task, running just `gulp` will compile the sass, launch BrowserSync & watch files.
  */
-gulp.task('default', ['dev']);
+// module.default = gulp.series('dev');
```
Comment (alex-deaconu) [inline gulpfile.js:351]: @ezequielc I commented this one out. In the original build `gulp` runs `dev`, but above `gulp` should be showing all the build steps. I'm not sure which one we want to stick with. Thanks.

Comment (ezequielc) [reply gulpfile.js:351]: @alex-deaconu I don't have a preference. In the README, it says that `gulp` will run local development. As long this is consistent with README, I am good with either. Thanks!

Comment (alex-deaconu) [reply gulpfile.js]: @ezequielc added a second `gulp info` task instead and kept `gulp` to run `dev`. Updated readme to reflect this. Thanks.

**Pull request merged**